packages/build-info/src/detect-build-system.ts

```diff
@@ -0,0 +1,154 @@
+      const pkgJson = getPkgJson(lernaConfigPath)
+      const { devDependencies } = pkgJson
+
+      return Promise.resolve({
+        name: 'lerna',
+        version: devDependencies?.lerna,
+      })
+    }
+  },
+
+  turbo: async (baseDir: string, rootDir?: string): Promise<BuildSystem | undefined> => {
+    const turbo = ['turbo.json']
+    const turboConfigPath = lookFor(turbo, baseDir, rootDir)
+
+    if (turboConfigPath) {
+      const pkgJson = getPkgJson(turboConfigPath)
+      const { devDependencies } = pkgJson
+
+      return Promise.resolve({
+        name: 'turbo',
+        version: devDependencies?.turbo,
+      })
+    }
+  },
+
+  rush: async (baseDir: string, rootDir?: string): Promise<BuildSystem | undefined> => {
+    const rush = ['rush.json']
+    const rushConfigPath = lookFor(rush, baseDir, rootDir)
+
+    if (rushConfigPath) {
+      const pkgJson = getPkgJson(rushConfigPath)
+      const { devDependencies } = pkgJson
+
+      return Promise.resolve({
+        name: 'rush',
+        version: devDependencies?.rush,
+      })
+    }
+  },
+
+  lage: async (baseDir: string, rootDir?: string): Promise<BuildSystem | undefined> => {
+    const lage = ['lage.config.js']
+    const lageConfigPath = lookFor(lage, baseDir, rootDir)
+
+    if (lageConfigPath) {
+      const pkgJson = getPkgJson(lageConfigPath)
+      const { devDependencies } = pkgJson
+
+      return Promise.resolve({
+        name: 'lage',
+        version: devDependencies?.lage,
+      })
+    }
+  },
+
+  pants: async (baseDir: string, rootDir: string): Promise<BuildSystem | undefined> => {
+    const pants = ['pants.toml']
+    const pantsConfigPath = lookFor(pants, baseDir, rootDir)
+
+    if (pantsConfigPath) {
+      return Promise.resolve({
+        name: 'pants',
+      })
+    }
+  },
+
+  buck: async (baseDir: string, rootDir: string): Promise<BuildSystem | undefined> => {
+    const buck = ['.buckconfig', 'BUCK']
+    const buckConfigPath = lookFor(buck, baseDir, rootDir)
+
+    if (buckConfigPath) {
+      return Promise.resolve({
+        name: 'buck',
+      })
+    }
+  },
+
+  gradle: async (baseDir: string, rootDir: string): Promise<BuildSystem | undefined> => {
+    const gradle = ['build.gradle']
+    const gradleConfigPath = lookFor(gradle, baseDir, rootDir)
+
+    if (gradleConfigPath) {
+      return Promise.resolve({
+        name: 'gradle',
+      })
+    }
+  },
+
+  bazel: async (baseDir: string, rootDir: string): Promise<BuildSystem | undefined> => {
+    const bazel = ['.bazelrc', 'WORKSPACE', 'WORKSPACE.bazel']
+    const bazelConfigPath = lookFor(bazel, baseDir, rootDir)
+
+    if (bazelConfigPath) {
+      return Promise.resolve({
+        name: 'bazel',
+      })
+    }
+  },
+}
+
+const lookFor = (configFile: string[], baseDir: string, rootDir?: string): string | undefined => {
+  return findUpSync(configFile, { cwd: baseDir, stopAt: rootDir })
+}
+
+const getPkgJson = (configPath: string): PackageJson => {
```

Comment:
we already have a package for that installed called `read-pkg` https://www.npmjs.com/package/read-pkg

This handles that a little bit more gracefully

Reply:
I opted to use `read-pkg` at first but it was causing my tests to fail because I am using the mockedFileSystem to mock the directories.